src/vs/workbench/parts/terminal/test/electron-browser/terminalLinkHandler.test.ts

```diff
@@ -47,47 +47,267 @@ class TestWorkspace implements IWorkspace {
 suite('Workbench - TerminalLinkHandler', () => {
 	suite('localLinkRegex', () => {
 		test('Windows', () => {
-			const regex = new TestTerminalLinkHandler(new TestXterm(), Platform.Windows, null, null).localLinkRegex;
-			function testLink(link: string) {
-				assert.equal(` ${link} `.match(regex)[1], link);
-				assert.equal(`:${link}:`.match(regex)[1], link);
-				assert.equal(`;${link};`.match(regex)[1], link);
-				assert.equal(`(${link})`.match(regex)[1], link);
+			const terminalLinkHandler = new TestTerminalLinkHandler(new TestXterm(), Platform.Windows, null, null, null);
+			function testLink(link: string, linkUrl: string, lineNo?: string, columnNo?: string) {
+				assert.equal(terminalLinkHandler.extractLinkUrl(link), linkUrl);
+				assert.equal(terminalLinkHandler.extractLinkUrl(`:${link}:`), linkUrl);
+				assert.equal(terminalLinkHandler.extractLinkUrl(`;${link};`), linkUrl);
+				assert.equal(terminalLinkHandler.extractLinkUrl(`(${link})`), linkUrl);
+
+				if (lineNo) {
+					const lineColumnInfo: LineColumnInfo = terminalLinkHandler.extractLineColumnInfo(link);
+					assert.equal(lineColumnInfo.lineNumber, lineNo);
+
+					if (columnNo) {
+						assert.equal(lineColumnInfo.columnNumber, columnNo);
+					}
+				}
 			}
-			testLink('c:\\foo');
-			testLink('c:/foo');
-			testLink('.\\foo');
-			testLink('./foo');
-			testLink('..\\foo');
-			testLink('../foo');
-			testLink('~\\foo');
-			testLink('~/foo');
-			testLink('c:/a/long/path');
-			testLink('c:\\a\\long\\path');
-			testLink('c:\\mixed/slash\\path');
-			testLink('a/relative/path');
+
+			testLink('c:\\foo', 'c:\\foo');
+			testLink('c:/foo', 'c:/foo');
+			testLink('.\\foo', '.\\foo');
+			testLink('./foo', './foo');
+			testLink('..\\foo', '..\\foo');
+			testLink('../foo', '../foo');
+			testLink('~\\foo', '~\\foo');
+			testLink('~/foo', '~/foo');
+			testLink('c:/a/long/path', 'c:/a/long/path');
+			testLink('c:\\a\\long\\path', 'c:\\a\\long\\path');
+			testLink('c:\\mixed/slash\\path', 'c:\\mixed/slash\\path');
+			testLink('a/relative/path', 'a/relative/path');
+
+			// With line and column number.
+			testLink('c:\\foo:5', 'c:\\foo', '5');
+			testLink('c:\\foo:5:3', 'c:\\foo', '5', '3');
+			testLink('c:\\foo:line 5', 'c:\\foo', '5');
+			testLink('c:\\foo:line 5, column 3', 'c:\\foo', '5', '3');
+			testLink('c:\\foo(5)', 'c:\\foo', '5');
+			testLink('c:\\foo(5,3)', 'c:\\foo', '5', '3');
+			testLink('c:\\foo (5)', 'c:\\foo', '5');
+			testLink('c:\\foo (5,3)', 'c:\\foo', '5', '3');
+			testLink('c:\\foo on line 5', 'c:\\foo', '5');
+			testLink('c:\\foo on line 5, column 3', 'c:\\foo', '5', '3');
+
+			testLink('c:/foo:5', 'c:/foo', '5');
+			testLink('c:/foo:5:3', 'c:/foo', '5', '3');
+			testLink('c:/foo:line 5', 'c:/foo', '5');
+			testLink('c:/foo:line 5, column 3', 'c:/foo', '5', '3');
+			testLink('c:/foo(5)', 'c:/foo', '5');
```

Comment:
It would be good to have example programs that output each format (and only add them if we have an example).